### ISSUE TITLE
WCS service with Tile-Index-Layer and Time support, overwrites FILTER on TileIndexLayer

### ISSUE THREAD

**Issue by ashleysommer** [MapServer/MapServer#6406]
## Expected behavior and actual behavior.
Expected behavior:
- Mapserver will append the Time filter to the existing filter in TileLayerIndex using msLayerSetTimeFilter
Actual behavior:
- Mapserver clears existing filter in TileLayerIndex before setting Time filter

See this paragraph about using TileLayerIndex for indexing on a WCS service:
https://mapserver.org/ogc/wcs_server.html#spatio-temporal-indexes
One of the benefits stated for using "new style" TileIndexLayer syntax, is you can set filters on the TileIndexLayer. But clearing the existing filter before apply msLayerSetTimeFilter negates the benefit.

## Steps to reproduce the problem.
Snippet of Mapfile:
```
LAYER
  NAME "MyTileIndex"
  TYPE TILEINDEX
  DATA "/path/to/my/index.shp"
  FILTER ( "[attrib]" eq "val" AND [attr2] ge 0 )
END
LAYER
  NAME "MyLayer"
  TYPE RASTER
  TILEINDEX "MyTileIndex"
  TILEITEM "location"
  PROJECTION
    "init=epsg:4326"
  END
  METADATA
    wcs_enable_request "*"
    wcs_label "MyLayer"
    wcs_rangeset_name "Range1"
    wcs_rangeset_label "mylayer"
    wcs_nativeformat "GeoTIFF"
    wcs_native_format "image/tiff"
    wcs_srs "EPSG:4326"
    wcs_formats "GeoTIFF"
    wcs_timeitem "time"
    wcs_timeposition "2015-11-20/2021-01-01/P1D"
  END
END
```

When querying the WCS service with TIME="2015-11-20", I expect the FILTER on MyTileIndex to become:
```
FILTER ((( "[attrib]" eq "val" AND [attr2] ge 0 )) AND (`[time]` = `2015-11-20`))
```
Instead, the existing filter is cleared, and the filter becomes:
```
FILTER ( `[time]` = `2015-11-20`)
```

See the code free's the exsting expression here:
https://github.com/MapServer/MapServer/blob/7c65ac4256d94e0dcc2870ef6f7ccb549f3b0f70/mapwcs.cpp#L1943-L1945
## Operating system
Ubuntu 20.04

## MapServer version and installation method
Mapserver 7.6.4

**Comment by ashleysommer** [comment]
I just noticed, even if msFreeExpression wasn't called at that location, it is called again in `msLayerSetTimeFilter()` because filter.type is `MS_EXPRESSION`, not `MS_STRING`:

https://github.com/MapServer/MapServer/blob/7c65ac4256d94e0dcc2870ef6f7ccb549f3b0f70/maplayer.c#L1426-L1434

**Comment by jratike80** [comment]
As a workaround you could filter the data with SQL. I guess that it requires OGR connection

```
CONNECTIONTYPE OGR
CONNECTION "/path/to/my/index.shp"
DATA "select * from index where attrib=val and attr2 >= 0"
```

**Comment by ashleysommer** [comment]
> As a workaround you could filter the data with SQL. I guess that it requires OGR connection
Yep, thanks for the suggestion. I've fixed the issue described above with #6408, but aside from that, I was going to look into doing SQL on an OGR connection as my next step. Good to know that will work if I need to go down that path.

**Comment by jratike80** [comment]
If you happen to have large and irregular coverages from a long time period and you just want to find the newest data that exist by using a very wide time range you will find that performance may be bad. See https://github.com/MapServer/MapServer/issues/5899.

We use a workaround by having a "best before" field in the tileindex. That is updated when there is newer data available for every pixel of the image.
